sars2seq/genome.py

```diff
@@ -211,16 +213,25 @@ def aaSequences(self, featureName):
         feature = self.features[featureName]
         name = feature['name']
 
-        referenceAa = AARead(
-            self.features.reference.id + f' ({name})',
-            feature.get('translation', translate(feature['sequence'], name)))
+        if (name == 'surface glycoprotein' and genomeNt.count('-') > 0 and
+                genomeNt.count('-') % 3 == 0):
+            referenceAaAligned = translateSpike(referenceNt)
+            genomeAaAligned = translateSpike(genomeNt)
 
-        genomeAa = AARead(
-            self.genome.id + f' ({name})',
-            translate(genomeNt.sequence.replace('-', ''), name))
+            assert len(referenceAaAligned) == genomeAaAligned, (
+                'Genome and reference AA sequences have different lengths.')
```

Comment:
You should add a test about the claimed raising of an `AssertionError`.

It looks to me like this code cannot possibly have been run. If you run `env PYTHONPATH=. trial --coverage test` and look in `_trial_temp/coverage/sars2seq.genome.cover` you will see this code is never touched by your tests (if it was it would raise an exception). See screenshot.

When I run the test suite on the branch of this pull request I get 31 errors.

<img width="843" alt="Screenshot 2021-06-19 at 23 05 04" src="https://user-images.githubusercontent.com/132978/122655488-cbb39700-d152-11eb-9668-6224530ad905.png">

Reply:
I played around and I can't see a way to actually get the assertion to raise. So I can either leave the assert just in case something weird ever happens or take it out.

Reply:
I left it in for now.